sufia-models/app/actors/sufia/generic_file/actor.rb

```diff
@@ -41,7 +41,7 @@ def create_content(file, file_name, path, mime_type, collection_id = nil)
     end
 
     def add_file_to_collection(collection_id)
-      return if collection_id.nil? || collection_id == -1
+      return if collection_id.nil? || ([-1, "-1"].include? collection_id)
```

Comment:
Do you find that this ID is sometimes passed as a string and other times as a fixnum?